azext_iot/operations/events3/_events.py

```diff
@@ -76,33 +77,56 @@ def _get_conn_props():
         properties["platform"] = sys.platform
         return properties
 
-    if not target.get('events'):
-        endpoint = _build_iothub_amqp_endpoint_from_target(target)
-        _, update = await evaluate_redirect(endpoint)
-        target['events'] = update['events']
+    if not target:
+        logger.debug('No Event Hub target found provided.')
+        return
+
+    if not target.get('central'):
+        if not target.get('events'):
+            endpoint = _build_iothub_amqp_endpoint_from_target(target)
+            _, update = await evaluate_redirect(endpoint)
+            target['events'] = update['events']
+            endpoint = target['events']['endpoint']
+            path = target['events']['path']
+            auth = _build_auth_container(target)
+            meta_data = await query_meta_data(target['events']['address'], target['events']['path'], auth)
+            partition_count = meta_data[b'partition_count']
+            partition_ids = []
+            for i in range(int(partition_count)):
+                partition_ids.append(str(i))
+            target['events']['partition_ids'] = partition_ids
+        else:
+            endpoint = target['events']['endpoint']
+            path = target['events']['path']
+        partitions = target['events']['partition_ids']
         auth = _build_auth_container(target)
-        meta_data = await query_meta_data(target['events']['address'], target['events']['path'], auth)
+    else:
```

Comment:
Could you spend some time optimizing this pattern. It is getting unwieldy + repeated code. Consider building a class/transformation component that intakes a `target` understands KPI's with respect to central or vanilla iothub and outputs an object whose attributes can easily be fed into the monitor_events co-routine?